tzatziki-spring/src/test/java/com/decathlon/tzatziki/steps/TestApplicationSteps.java

```diff
@@ -35,4 +48,28 @@ public void before() {
     public void if_we_disable_the_http_interceptor() {
         HttpInterceptor.disable();
     }
+
+
+    @Given(Patterns.THAT + "clean thread pool executor is (enabled|disabled)")
+    public void clead_thread_pool_executor(String enabled) {
+        SpringSteps.clearThreadPoolExecutor = "enabled".equals(enabled);
+
+    }
+
+    @Given(Patterns.THAT + "we start an infinite task")
+    public void start_infinite_task() {
+        completableFutureToTest = taskExecutor.submit(() -> {
+            try {
+                new CountDownLatch(1).await(10, TimeUnit.SECONDS);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        });
```

Comment:
Perhaps put the await to 1000 seconds to ensure it is really not waiting for 10 sec ?
Also, be aware that it is working there only because an InterruptedException pops out when the shutdown call is done.